File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, n: int) -> bool:
        copy = n
        
        # After the code, revNum will
        # contain the reversed number
        revNum = 0

        # Keep on iterating while there
        # are digits left to extract
        while n > 0:
            lastDigit = n % 10

            # Pushing last digit at the
            # back of reversed number
            revNum = (revNum * 10) + lastDigit
            n = n // 10
        
        # Return true if the reversed and 
        # copy of original number is same
        return revNum == copy

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        int revNum = 0;
        int dummy = x;
        while (dummy > 0) {
            int digit = dummy % 10;

            revNum = (revNum * 10) + digit;
            dummy = dummy / 10;
        }
        return revNum == x;
    }
}